### DISCUSSION TITLE
Setting a button enable in relation to data in budibase DB

### DISCUSSION THREAD

**Discussion by proton029** [Budibase/budibase#6884] [Help] [answered]
Hello all.
I have a database(budibase DB). My form has a single txt field of Email and a search button . I want to do something like when i enter an email and press a search  i want it to check whether that email is present in DB and if not enable a button which links to another screen. Yes  there is filter option which we can configure search only email. but I want it to be a text field. How can i do this ?  :)

**Comment by melohagan** [comment, marked as answer]
Hello @proton029 

What you can try is have a Data Provider with a filter: `email EQUALS {{ Form.TextField }}`

Then you can have a <a href="https://docs.budibase.com/docs/conditions">condition</a> on your button that either **hides** or **update setting** to *disabled* if `{{ Data Provider.Rows length }}` more than 0

**Reply by proton029** [reply]
Hi @melohagan 
Thanks for the reply the method did work . Now I'm facing a new problem that the emails that are already set as primary key are getting duplicated if a new form is submitted in that id. Can we stop this using some validation during submission of the form?

**Reply by melohagan** [reply]
It would be very similar to the condition, but instead you would use the **Continue if / Stop if** action.

E.g.
![Screenshot 2022-07-26 at 10 27 48](https://user-images.githubusercontent.com/101575380/180973395-e8cc7bea-adf0-4254-a965-789cac22a9fd.png)

In this case the Data Provider would have a filter for the email address you want to search for.
Therefore, IF email is not in db THEN continue to Save Row.

Otherwise the save will not happen.

**Reply by proton029** [reply]
Thankyou so much @melohagan
